Kkachmar/remove session start metric

kekachmar opened the pull request:
This change removes the SessionStartMetric from Scribe and Deli. The reasons behind this change are:
1. There has been a bug where the size of the checkpoints is too large, causing checkpoints to fail.
2. A likely cause of this is the `successfullyStartedLambdas` data object, which can be very large, and is used in the `SessionStartMetric`. 
3. When checkpoints fail, the `successfullyStartedLambdas` value in the checkpoint could be incorrect, which may lead to invalid metrics.
4. With updated logs and metrics, we don't need this metric or the validation of `successfullyStartedLambdas`, as we have logging around session start behaviors such as the `restoreFromCheckpoint` metric and `RunService` metric, as well as logging for creating document partitions for document lambdas like Scribe and Deli.

Josmithr committed:
refactor(telemetry-generator): Upgrade eslint config to "recommended" and fix violations in code (#21076)

This package was using the `minimal-deprecated` configuration, which is
scheduled for deletion. Upgrades the package's config to the
`recommended` base and fixes violations in the code.

This PR disables some of the `recommended` rules due to the sheer number
of violations. These can and should be removed and fixed in the future.
For now, the primary goal here is to remove usages of the deprecated
config.


[AB#2900](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/2900)

---------

Co-authored-by: Alex Villarreal <716334+alexvy86@users.noreply.github.com>

Josmithr committed:
refactor(getKeys): Upgrade eslint config to "recommended" and fix violations in code (#21078)

This package was using the `minimal-deprecated` configuration, which is
scheduled for deletion. Upgrades the package's config to the
`recommended` base and fixes violations in the code.

This PR disables some of the `recommended` rules that are
TypeScript-specific. Longer term, we should create a better pattern for
JavaScript packages. For now, the primary goal here is to remove usages
of the deprecated config.

AB#8051

shubhi1092 committed:
Skip document corruption if partition is closed (#21077)

## Description

Adding a check in doc corruption logic to skip marking the doc as
corrupted if the doc partition is already closed (eg: if the session is
paused)

This will help reduce unnecessary doc corruptions in case of paused
sessions as well as reduce uncaught exception restarts due to emitting
error for a paused session.

---------

Co-authored-by: Shubhangi Agarwal <shuagarwal@microsoft.com>

chentong7 committed:
refactor(container-runtime): Tighten up exposure of ContainerRuntime's logger (#20839)

## Description

ContainerRuntime.logger is currently public, though ideally, it should
not be, and it uses an internal type instead of the preferable base
type. When sharing the logger, the standard practice should be to pass
around the base logger, allowing each consumer to create a ChildLogger
from it.

## Changes

* Rename ContainerRuntime.logger to ContainerRuntime.baseLogger, change
its type to ITelemetryBaseLogger, and set its access level to private.

* Adjust the dependencies that receive the logger to accept the
ITelemetryBaseLogger type and utilize createChildLogger to create a
child logger as needed (this operation will be a no-op if the logger is
already a child logger).

* Modify the following classes in Fluid Framework (FF) so they no longer
directly access the logger from the runtime but instead receive their
own base logger via constructor parameters:

BlobManager
Summarizer
DataStoreContext

* DataStoreContext currently makes the same logger available through its
public API. This may be unnecessary—as indicated by its lack of use in
the FF codebase—especially since IFluidDataStoreRuntime already exposes
a logger. This exposure could potentially be removed to streamline the
API and enhance encapsulation.

CraigMacomber committed:
Fix type test symbol handling (#21070)

## Description

Improve type test symbols handling. Well known symbols are preserved, as
is `symbol`, but unique symbols are replaced (with never), both when
used as keys and as value types preventing valse positives (detected API
breaks that are op-ops).

This was manually tested with the RC4 client packages, a determined to
fix the issue with the new fluid handle symbol.

agarwal-navin committed:
Make the new summary retry logic the default and remove old logic (#20851)

https://github.com/microsoft/FluidFramework/pull/16885 added a new retry
mechanism in case of summarization failures. This was rolled out with
A/B testing and the experiment was successful. This PR makes the new
retry mechanism default and removes the old one.

**Notes**
- Added retry for these two additional scenarios - When a summary op or
summary ack is not received within the timeout, retry. This is because
these failures are mostly transient and a retry may fix them.
- Removed `refreshLatestAck` summary option because its not used
anymore. A lot of the code changes are because of this.


[AB#7605](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7605)

ChumpChief committed:
Use post-creation resolved URL to handle discovery pattern (#21080)

alexvy86 committed:
docs(devtools): Update landing page text (#21084)

yann-achard-MS committed:
docs(tree): Update undo and detach trees docs (#20754)

CraigMacomber committed:
Update build-tools and type tests (#21087)

## Description

https://github.com/microsoft/FluidFramework/pull/21070 made some
improvements to type tests which were needed to regenerate the type
tests for client. This change integrates that, as well as regenerates
the type tests with the new generator.

Toritos01 committed:
E2E ephemeral tests with manual payloads (#20874)

## Description

- This PR adds the capability for the azure client tests to run against
ephemeral containers when the azure__fluid__relay__service__ephemeral
environment variable is set to "true"
- This is done without any changes to the AzureClient by running the
ephemeral tests using manually created Axios requests with pre-generated
summary trees (stored in [ephemeralSummaryTrees.ts)
- The pre-generated summary trees were obtained by running a debugger
against the existing E2E tests, extracting the summaries created by the
AzureClient, and then modifying the IsEphemeralContainer property of
these objects
- Certain tests are skipped in ephemeral mode, because this method of
creating containers does not involve attachment/detachment logic
- Non-ephemeral E2E test behavior should remain unchanged


- Most of the file changes not in the packages/service-clients folder
are just some autogenerated file changes

---------

Co-authored-by: Brandon Diaz <“BrandonLouisDiaz@gmail.com”>
Co-authored-by: Matt Rakow <ChumpChief@users.noreply.github.com>

Josmithr committed:
refactor(tree): Enable explicit return type rule and fix violations (#21066)

In preparation for updating the package to use our recommended base
eslint config, this PR enabled the `explicit-return-types` rule in the
package (excluding test code) and fixes violations (while disabling the
rule in select cases for functions that intentionally derive their
return type from other functions/libraries).

AB#6983

andrewstanovsky committed:
Add OAuthBearer authentication support for rdkafka (#21058)

## Description

Adds support for OAuthBearer authentication flow for rdkafka client when
Kafka server expects/supports it. The 'tokenProvider' is supposed to be
set via nconf programatically.

## Breaking Changes


server/routerlicious/packages/services-ordering-rdkafka/src/rdkafkaBase.ts
introduces connect() method breaking change by changing its return value
from 'void' to 'Promise<void>'. This is needed for tokenProvider's call
convenience and overall maintainability/readability of the code.

## Reviewer Guidance

This change has been verified with FRS and three types of Kafka server:

* HDI Kafka w/ SSL auth
* Event Hubs w/ connection string
* Event Hubs w/ managed identity


Another change that is not directly related to the main topic is fix for
incorrect node-rdkafka's feature filtering
```
		const rdKafkaHasSSLEnabled = kafka.features.filter((feature) =>
			feature.toLowerCase().includes("ssl"),
		);
```
The `rdKafkaHasSSLEnabled` will have an empty array if `ssl` is not
included. Which later would be checked with the following line:
```
		if (!rdKafkaHasSSLEnabled) {
```
The problem with this line is that it will always be false whether
`rdKafkaHasSSLEnabled` contains `ssl` entry or it's completely empty.
Empty array is still not null/undefined, so the condition will be false.

I've changed it to an array of lower case values from the original one
of rdkafka (redundant step, since it always contains lower case
features, but to make sure it's always lower case and it doesn't change
the previous logic).
```
		const lcKafkaFeatures = kafka.features.map((s) => s.toLowerCase());
```

And new way of checking it traverses this new array and checks it with
the `includes()` method:

```
		if (!lcKafkaFeatures.includes("ssl")) {
```

markfields committed:
Offline: Minor refactoring of stashedOps.spec.ts (#21052)

Reading through stashedOps.spec.ts, I wanted to make a couple changes:

* Move helpers out of the top-level `describe` block, because they were
inconsistently using both "global" state from that scope and objects
passed in via parameters
* Moved `waitForSummary` helper out too, so callsites are explicit about
which container is being awaited
* I added comments and tweaked some names/strings in the test `handles
stashed ops created on top of sequenced local ops` after stepping
through it, it was a very interesting one to follow in terms of
learning.

Bonus change: Move event-specific logging props into the "details"
property. It's easier to find this info that way (you don't have to
remember the exact name), and avoids an explosion of columns when the
telemetry sink turns each prop into its own column.

RishhiB committed:
Test Update to Arrow application to v4 for secretless auth (#21075)

cherry pick 3073937895c60ee071228fef014c732d90f86687 to test if updating
arrow app to v4 will break the pipelines

[AB#7557](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7557)

Abe27342 committed:
Use node's bundled CA on windows (#21096)

## Description

Use node's bundled CA on windows.

Co-authored-by: Abram Sanderson <absander@microsoft.com>

kekachmar committed:
remove session start metric

alexvy86 committed:
feat(tree): Schema validation (#21011)

Abe27342 committed:
Use public client auth for odsp e2e flows (#21091)

## Description

Currently, we use the ropc flow with a confidential client to
authenticate in our e2e/stress tests against odsp. This is less
appropriate than using a public client flow, since our application
registration really only needs to have delegated permissions.
This PR adjusts things to use a public flow--I've updated the relevant
app registrations to allow both forms of auth already.

Using a public flow also means our infrastructure setup here aligns
exactly with [this relatively recent MSFT
guidance](https://learn.microsoft.com/en-us/entra/identity-platform/test-automate-integration-testing?tabs=dotnet).

---------

Co-authored-by: Abram Sanderson <absander@microsoft.com>

WayneFerrao committed:
refactor(devtools): Abstracted all  Menu-related logic to its own file (#21097)

## Description

This PR moves all code related to Menu into Menu.tsx. This was done to
clean up and organize the code in order to ease future accessibility
testing

RishhiB committed:
update remaining arrow app service connections (#21103)

update remaining arrow service connection mentions in code base

yann-achard-MS committed:
fix(tree): throw UsageError on invalid move (#21100)

sonalideshpandemsft committed:
Change ARROW property name (#21104)

build-docs pipeline fails as the property name in not updated to
`ArrowARMServiceConnection` here:
https://github.com/microsoft/FluidFramework/pull/21103

jatgarg committed:
Respect the odspEndpoint argument if supplied in e2e tests. (#21105)

## Description

E2E test does not respect the odspEndpointName argument is supplied. If
not supplied, they run against odsp PROD. Fix it to respect the
argument.

---------

Co-authored-by: Jatin Garg <jatingarg@Jatins-MacBook-Pro-2.local>

vladsud committed:
Better telemetry for disconnect events: record pendingClientId (#21093)

When connection does not move far enough (does not get to "Connected" state), we do not have good mechanism to pair "establishing connection" events with "disconnect" events (unless complex join queries are involved), as clientId can't be used for such pairing. We need pendingClientId for that, but it's reset on disconnect before telemetry event fires.

This change moves the time when we reset it, such that it gets into telemetry.

scottn12 committed:
feat(azure-end-to-end-tests): Expand AzureClient e2e tests (#21041)

## Description

This PR updates `@fluidframework/azure-end-to-end-tests` to import
legacy (LTS) AzureClient APIs. This will allow us to test collaboration
between legacy and current AzureClient users.

Additionally, this PR adds AzureClient e2e tests (against tinylicious)
to the build pipeline for PRs. This will require AzureClient tests to
pass to check-in future PRs.

## Reviewer Guidance
- Additional 1.x/2.x compatibility smoke tests that you would like to
see added in this or a future PR
- Thoughts on the method I used to have Azure e2e tests ran in build
pipelines for PRs (see the
[logs](https://dev.azure.com/fluidframework/public/_build/results?buildId=263659&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3&t=ddd9efee-bc1d-59c0-82a9-543538b126ce)
for this PR as an example).
- Thoughts on importing the legacy AzureClient/SharedMap APIs directly
in package.json. The alternative is to setup a programmatic way to
download these legacy packages (more similar to the setup in
`@fluid-private/test-end-to-end-tests`)

## Misc

[One-pager](https://microsoft.sharepoint-df.com/:fl:/g/contentstorage/CSP_5fcca107-c97c-4b0a-899e-42a2578d1609/EX3Cew9vjshMmgeYPckgHHkB7O_6yukNZg00D2mrpR2kMA?e=ZBWefW&nav=cz0lMkZjb250ZW50c3RvcmFnZSUyRkNTUF81ZmNjYTEwNy1jOTdjLTRiMGEtODk5ZS00MmEyNTc4ZDE2MDkmZD1iJTIxQjZITVgzekpDa3VKbmtLaVY0MFdDY3JoRTNub2dqNURqZ01yWHlBTGQ3MWxiZ0VwY3VvYlFKXzhnb2luWjJFTyZmPTAxTUdUSktNTDVZSjVRNjM0T1pCR0pVQjRZSFhFU0FIRFomYz0lMkYmYT1Mb29wQXBwJnA9JTQwZmx1aWR4JTJGbG9vcC1wYWdlLWNvbnRhaW5lciZ4PSU3QiUyMnclMjIlM0ElMjJUMFJUVUh4dGFXTnliM052Wm5RdWMyaGhjbVZ3YjJsdWRDMWtaaTVqYjIxOFlpRkNOa2hOV0RONlNrTnJkVXB1YTB0cFZqUXdWME5qY21oRk0yNXZaMm8xUkdwblRYSlllVUZNWkRjeGJHSm5SWEJqZFc5aVVVcGZPR2R2YVc1YU1rVlBmREF4VFVkVVNrdE5UMFZDTWxGRFRVSlZSMVpDUVV4R1RGVkpNelpOTlVaT1JqTSUzRCUyMiUyQyUyMmklMjIlM0ElMjJhNDE5Zjk2MS1mNDA2LTQ5MmEtYmZjNi0yZTFmYTFiYzIyNjklMjIlN0Q%3D)
for context

[AB#7690](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7690)

taylorsw04 committed:
refactor(id-compressor): Cleanup compressor usage in runtime (#21082)

## Description

This PR cleans up the runtime/outbox usage of the ID compressor.

agarwal-navin committed:
[GC] Remove the option to disable GC mark phase (#21101)

GC is enabled by default and has been this way for quite a while.
Specifically, the mark phase which marks unreferenced objects as such in
the summary is always enabled. This does not effect the container's
content or its usage in any way meaning that it doesn't put any
restrictions on content retrievel and it doesn't delete anything. These
are done by the other two stages of GC - tombstone and sweep. These can
still be controlled via settings.

This PR removes the options to turn GC mark phase off. It will always be
enabled and cannot be turned off:
- `gcAllowed` option was used to control the mark phase roll-out. The
roll out was completed couple of years ago and has been stable. This
option is now removed.
- `disableGC` option was to provide ability to turn it off in case
things went wrong. This was never used (except in tests) and is mow
removed.
- The mark phase can still be turned off in tests via the feature flag
`Fluid.GarbageCollection.Test.RunGC`. This used to be called
`Fluid.GarbageCollection.RunGC` and is renamed in this PR to include
`Test`.

jason-ha committed:
fix(build-tools): fluid-build-tasks-eslint deps (#21089)

getDefaultTscTaskDependencies was ineffective and is now removed. eslint
with ts processing does not require build first. Instead of requiring
that tsc is run for the projects listed in lint config, depend on the
things the respective tsc tasks depend on.

Once policy changes are picked up by client group ^tsc dependencies
listed for eslint in examples/data-objects can be removed.

Abe27342 committed:
Remove plumbing of app secret for odspdf stress (#21113)

## Description

Removes some extra references for plumbing the environment variable
`login__microsoft__secret` in stress tests. Follow-up to #21091.

Co-authored-by: Abram Sanderson <absander@microsoft.com>

kian-thompson committed:
Add performance events for test-service-load client election (#21086)

We are seeing weird gaps in summarizer in test-service-load runs. This
PR adds optional performance events (only enabled in test-service-load)
to track adding, removing, and electing clients.

scottn12 committed:
fix(build): Add ci:test:azureclient:tinylicious:coverage command (#21117)

## Description

Add missing command for AzureClient e2e tests when running post check-in
build.

yann-achard-MS committed:
refactor(tree): prepare for new sequence field changeset format (#21120)

scottn12 committed:
fix(azure-end-to-end-tests): Include AzureClient e2e tests in ci:test:realsvc:tinylicious (#21121)

## Description

This PR makes it so AzureClient e2e tests are ran as part of the
`ci:test:realsvc:tinylicious` command instead of having its own phase.

kekachmar committed:
remove session start metric from deli lambda factory

Abe27342 committed:
docs(tree): Add example PRs for making backward-compatible format changes (#20617)

## Description

Adds an example to the docs for making backward-compatible format
changes in tree.

---------

Co-authored-by: Abram Sanderson <absander@microsoft.com>

agarwal-navin committed:
[Summarization] Added type for the error thrown during summarization (#21085)

The error thrown during summarization go through multiple layers -
container runtime -> summary generator -> running summarizer. Currently,
the error type is `Error` and it gets lost while is moves through these
layers.

This change fixes this by improving the types of the summarizer errors
through these layers. It does the following:
- Renames `IRetriableFailureResult` -> `IRetriableFailureResult` and
uses it consistently as the error type throughtout all these layers.
- In SummaryGenerator / RunningSummarizer, ensures that all failures
have an error associated with them.
- Removes the error from the failure results which was a duplicate of
the error in the base result.

nmsimons committed:
Add readme to Example folder (#21119)

Adding a readme file that warns people not to use these examples as
examples of how to use Fluid Framework.

CraigMacomber committed:
Remove format validation from summary load benchmark (#21127)

## Description

Looking at the perf benchmarks showed a big increase in summary load
performance when updating typebox. This is mostly likely during the
optional data validation which we don't expect production users to
enable (and don't even provide a stable APi to use).

This change removes this validation from the summary load benchmarks so
they should better reflect realistic use.

alexvy86 committed:
fix(e2e-tests): Improve handling of use-openssl-ca when running r11s e2e tests (#21133)

markfields committed:
Offline: Add negative tests to illustrate gap in capability around identifying ops submitted in multiple incarnations of the same container (#21095)

We need to handle the case where a container is
serialized, and then multiple incarnations of the same container
resubmit the same ops. Those resubmitted ops will all be on unique
clientIds, so the existing op ID logic (which just identifies "local"
ops as those matching the current clientId) will not recognize them as
duplicates.

This PR adds 3 test cases illustrating this gap.

yunho-microsoft committed:
Reduce session grace period for ephemeral containers when cluster is under draining (#21010)

Set the session activity timer to a shorter value if the current session
is ephemeral and cluster is under draining. It will make session end
quicker than usual (normally 10 minutes)

---------

Co-authored-by: Yunpeng Hou <yunho-microsoft@YunhoMakPro2019.guest.corp.microsoft.com>

CraigMacomber committed:
Type Erase Shared object kinds (#21081)

## Description

Make ISharedObjectKind, IChannel, IChannelFactory,
IFluidDataStoreRuntime and more alpha by tweaking fluid-static APIs
typing.

## Breaking Changes

Access to these now less public types should not be required for users
of the `@public` "declarative API" exposed in the `fluid-framework`
package, but can still be accessed for those who need them under the
`/legacy` import paths.
The full list of such types is:

- `SharedTree` as exported from `@fluidframwork/tree`: It is still
exported as `@public` from `fluid-framework` as `SharedObjectKind`.
- `ISharedObjectKind`: See new `SharedObjectKind` type for use in
`@public` APIs.
    `ISharedObject`
-   `IChannel`
-   `IChannelAttributes`
-   `IChannelFactory`
-   `IExperimentalIncrementalSummaryContext`
-   `IGarbageCollectionData`
-   `ISummaryStats`
-   `ISummaryTreeWithStats`
-   `ITelemetryContext`
-   `IDeltaManagerErased`
-   `IFluidDataStoreRuntimeEvents`
-   `IFluidHandleContext`
-   `IProvideFluidHandleContext`

Removed APIs:

-   `DataObjectClass`: Usages replaced with `SharedObjectKind`.
-   `LoadableObjectClass`: Replaced with `SharedObjectKind`.
- `LoadableObjectClassRecord`: Replaced with `Record<string,
SharedObjectKind>`.

yann-achard-MS committed:
refactor(tree): throw usage errors on invalid revertible usage (#21137)

yann-achard-MS committed:
refactor(tree): lineage-free format for sequence field changesets (#21131)

CraigMacomber committed:
Normalize imports (#21140)

## Description

Runs: `pnpm flub exec -g client --skipScope "@types" -- "pnpm flub
modify fluid-imports --onlyInternal --tsconfigs ./tsconfig.json
./src/test/tsconfig.json"`

Replace `from "@fluid-experimental/odsp-client/internal";` with `from
"@fluid-experimental/odsp-client";` to mitigate linter errors and run
`pnpm run format`.

zhangxin511 commented on server/routerlicious/packages/lambdas/src/deli/lambda.ts:
```diff
@@ -299,7 +293,6 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
 		private readonly rawDeltasProducer: IProducer,
 		private readonly serviceConfiguration: IServiceConfiguration,
 		private sessionMetric: Lumber<LumberEventName.SessionResult> | undefined,
-		private readonly sessionStartMetric: Lumber<LumberEventName.StartSessionResult> | undefined,
```
Why remove this metric? I think reliability might use this?

kekachmar replied:
Hi @zhangxin511, based on the details above, it is likely not providing an accurate reliability measurement. @pradeepvairamani thought it would be best to remove this. We I discussed that we have better ways of measuring reliability, such as `RunSession` metrics for the individual lambdas.

kekachmar replied:
@pradeepvairamani do you have any additional reasons?

pradeepvairamani replied:
The main reason is that this metric is just a consolidation of deli start and scribe start and to get this telemetry info, we are also introducing a control op and writing to checkpoint which all makes this quite expensive to just get this telemetry info. We already bubble up the errors during lambda start, but if we want we can add individual lambda start metrics which should be much cleaner.

zhangxin511 replied:
Thank you @kekachmar and @pradeepvairamani . I was just confirming. With this being said, is there a metric we could use to replace for this?
![image](https://github.com/microsoft/FluidFramework/assets/3719246/a1852f49-ef1c-45c7-96e7-6deadb8c7c7b)

kekachmar replied:
@zhangxin511 / @pradeepvairamani, do we want to address this further and add individual lambda start metrics?

zhangxin511 replied:
@kekachmar , we can build this metric later .

zhangxin511 commented on server/routerlicious/packages/lambdas/src/deli/lambda.ts:
```diff
@@ -816,47 +800,6 @@ export class DeliLambda extends TypedEventEmitter<IDeliLambdaEvents> implements
-		);
-	}
-
 	private logSessionEndMetrics(closeType: LambdaCloseType) {
```
So do we don't log start but still log end?

Josmithr committed:
refactor(tree): Enable `no-explicit-any` eslint rule and fix violations in code (#21098)

I fixed what was reasonably trivial. For cases that required deeper
investigation or that would affect the public API surface, I disabled
the rule. We can follow up on cleaning up these existing violations in
the future - for now, the rule will prevent new usages.

CraigMacomber committed:
Add ISharedSegmentSequence (#21067)

## Description

Fix that ISharedString extends SharedObject (the class).

This resolves one issue preventing tagging of "legacy" DDSes as public
since before this change doing so for SharedString would pull in a lot
more than it should.

## Breaking Changes

ISharedString no longer extends SharedSegmentSequence and instead
extends the new ISharedSegmentSequence, which may be missing some APIs.

Attempt to migrate off the missing APIs, but if that is not practical,
request the be added to ISharedSegmentSequence and cast to
SharedSegmentSequence as a workaround temporally.

WayneFerrao committed:
fix(devtools): Fixed accessibility issues in navigating Menu, Data visualizer and ev… (#21138)

[AB#7791](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7791)

## Description

This PR fixes accessibility issues with navigating/interacting with
various components in DevTools. The fixes include:
- Navigating through the menu
- Moving event filters out of the table so they're
focusable/interactable
- Navigating through the Data Visualizer


https://github.com/microsoft/FluidFramework/assets/23732584/784f90cf-3808-4672-97d8-f0c82672e6b0

kian-thompson committed:
[Service load test] Force some connections to be write mode for Tinylicious (#21136)

We are seeing a large number of "OpRoundtripTime" errors for Tinylicious
runs, largely caused by replaying ops at the beginning of the run.
Forcing some of these 120 clients to be in write mode from the start
should help alleviate some of the load.

After this change is merged, I will be monitoring the CI runs to see if
the error rate improves. If not, I'll revert this change.


[AB#7819](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7819)

CraigMacomber committed:
Remove several types from `@public` scope (#21142)

## Description

Due to my previous changes to make IFluidDataStoreRuntime non-public,
several types no longer need to be public. This moves more of them to
alpha.

Also includes a couple of `IAudience` related tweaks to make the types
better convey what I I think is supported as far as method binding and
modification.

## Breaking Changes

The following types have been moved from `@public` to `@alpha`:

-   `IFluidSerializer`
-   `ISharedObjectEvents`
-   `IChannelServices`
-   `IChannelStorageService`
-   `IDeltaConnection`
-   `IDeltaHandler`

These should not be needed by users of the declarative API, which is
what `@public` is targeting.

sonalideshpandemsft committed:
Upload manifest file for release branch (#21143)

The manifest file is uploaded by date and build number for main branch.
This change extends the functionality for release branches too.
 
For release branches, the manifest file is uploaded based on the build
number, rather than the date. This approach prevents it from overriding
the one uploaded from the main branch.
 
Once this PR is merged in main, I can port the change to release
branches.

Abe27342 committed:
test(tree): Snapshot test all SharedTreeWriteFormat options (#20644)

## Description

Refactors the two top-level SharedTree snapshot tests (for op format and
summary format in various scenarios) to snapshot test SharedTree using
all possible values of `SharedTreeFormatVersion`. This allows
scrutinization of new format options at the time they become public via
PR, and also some minimal validation (through scenarios not crashing)
that they can be used.

---------

Co-authored-by: Abram Sanderson <absander@microsoft.com>

Josmithr committed:
tools(changelog-generator-wrapper): Upgrade eslint config to "recommended" (#21151)

The minimal config is deprecated and scheduled for deletion soon.

Toritos01 committed:
Fix undefined command name error in Redis Client (#21158)

## Description
Small change to fix an error where error.command was undefined

Co-authored-by: Brandon Diaz <“BrandonLouisDiaz@gmail.com”>

WillieHabi committed:
Delete yaml file for Fluid AFR and Reliability tests (#21145)

The Fluid AFR Perf and Reliability Tests pipelines are no longer
monitored. This is the first step before removing the pipeline and
cleaning up it's env variables from key-vault.

https://dev.azure.com/fluidframework/internal/_workitems/edit/7997

CraigMacomber committed:
Cleanup `fluid-framework` legacy exports (#21153)

## Description

Cleanup `fluid-framework` legacy exports to remove no longer required
types.

## Breaking Changes

Removed types should not be needed for declarative API users, but can
still be found in their original packages.

jason-ha committed:
docs(client): comment on f-f API organization (#21149)

+ fix pending lint defects

Josmithr committed:
refactor(lambdas): Upgrade eslint config to "recommended" and fix violations in code (#21155)

The minimal config is deprecated and scheduled for deletion. Updates the
package to use the "recommended" config and fixes violations in code.

Disabled a number of rules in the config due to the sheer number of
violations. These can (and should) be fixed at a later date.

[AB#8002](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8002)

CraigMacomber committed:
Update client to TypeScript 5.3.3 (#20994)

## Description

Typescript 5.3 brings useful features like support for
[Symbol.hasInstance
narrowing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html#instanceof-narrowing-through-symbolhasinstance),
type checking optimizations and much more. Since ClientRequirements.md
lists our requirement at TypeScript 5.4, this should be ok (from the
perspective of aligning with that requirement)

See
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-2.html
and
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-3.html
for more details.

Based on a two test runs, this change makes `pnpm run build:compile`
about 7% faster. (105.438 and 107.011 seconds instead of 113.816 and
114.096 without the change)

## Breaking Changes

This is not expected to break anything, however its possible that the
compile TypeScript is slightly different, which could cause issues in
some edge cases.

This will enable code to start relying on newer features, which when
used in public APIs could in turn cause breakages for users of older
versions of TypeScript.

There are a couple cases where type-tests were impacted. These seemed to
be a fix of typescript capturing more accurate types, especially around
index signatures, and cases where the TypeScript compiler wouldn't
provide any useful information about what changed with the only
difference being inside the `...` non-printed parts of the types. As the
changes seem relatively minor they seem unlikely to cause any real
issues (didn't break lots of examples/tests) and since we only support
customers using TypeScript 5.4 and plan to move to 5.4 ourselves at some
point, if this causes a breaking change in some edge case, that probably
a good thing to get out of the way now while we are still in the RC
phase.

markfields committed:
Make handle parsing/decoding functions idempotent (#21102)

Fixes #8016. We need to port this back to RC3 and RC4 once merged to
main.

For anyone implementing their own DDS, when they pick up an FF release
with #19242 in it, they will likely be in a position where handles are
going through `parseHandles` or `serializer.decode` twice.

We assumed this was idempotent but found that it is not, with two
different codepaths needing to be fixed.

Note that `parseHandles` may return the same object unmodified now,
wherease before it would always clone (via `JSON.stringify` roundtrip)

---------

Co-authored-by: Alex Villarreal <716334+alexvy86@users.noreply.github.com>

CraigMacomber committed:
Tree: Fix TreeViewEvents.afterBatch (#21162)

## Description

`TreeViewEvents.afterBatch` is now triggered when appropriate instead of
never firing.

Toritos01 committed:
fixing Redis client command name bug in historian (#21163)

## Description
Pulling change from R11s that fixes redis command name bug

---------

Co-authored-by: Brandon Diaz <“BrandonLouisDiaz@gmail.com”>

jatgarg committed:
Fix parsing of GroupId in snapshotTree in odsp driver (#21164)

## Description

Fix parsing of GroupId in snapshotTree in odsp driver. We need to get it
from child and put it on child node.

Co-authored-by: Jatin Garg <jatingarg@Jatins-MacBook-Pro-2.local>

tyler-cai-microsoft committed:
Fix snapshotTree.id bug in driver (#21160)

[AB#8050](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8050)

Fix a contract violation detected by the `serializedStateManager`.
Essentially, we were not propagating the snapshot.id in the driver. Now
the snapshot version will contain the snapshot.id and things will work
as expected.

tyler-cai-microsoft committed:
Data Virtualization: Supports control over odsp cache in e2e tests (#21161)

[AB#8051](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8051)

This enables us to support controlling of the odsp cache so that we can
control what snapshots we load from in the e2e tests.

yann-achard-MS committed:
refactor(tree) use separate cells ids for src and dst cells in moves (#21156)

Josmithr committed:
docs(map, tree): Explicitly call out in SharedMap and SharedTree map node APIs that no enumeration ordering is guaranteed (#21146)

In both of these APIs, we attempt to mirror the JavaScript Map APIs,
which guarantee enumeration ordering matches insertion order. Neither
SharedMap nor SharedTree make such guarantees. This PR adds
documentation to make this clear to the end-user.


[AB#7955](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/7955)

CraigMacomber committed:
tree: Fix reading tree during events (#21172)

## Description

Fix bug where reading tree during events could cause issues.

Reading the tree inside of NodeChange and TreeChange events could
corrupt internal memory structures leading to invalid data in
subsequence reads as well as internal errors being thrown. This bug has
been fixed.

CraigMacomber committed:
Tree: Align tree tsconfigs with repo conventions (#21165)

## Description

Update how tree consumes our base configs (based on how the cell backed
does it as it seems like a clean up to date example).
Remove redundant options not needed due to values from base config.

kekachmar committed:
Merge branch 'main' into kkachmar/removeSessionStartMetric

kekachmar committed:
fix merge conflict

alexvy86 reviewed (approved):
Approving on behalf of devtools so the PR doesn't have to be recreated, but didn't look in detail at the changes.
